lib/voxpupuli/release/rake_tasks.rb

```diff
@@ -52,3 +52,90 @@
     fail "Unable to find a CHANGELOG.md entry for the #{v} release."
   end
 end
+
+desc "Prepare a release"
+task "release:prepare" do
+  Rake::Task["release:porcelain:changelog"].invoke
+  Rake::Task["strings:generate:reference"].invoke if File.exist?('REFERENCE.md')
+end
+
+begin
+  require 'github_changelog_generator/task'
+  require 'puppet_blacksmith'
+
+  class GCGConfig
+    def self.user=(user)
+      @user = user
+    end
+
+    def self.user
+      @user || metadata['name'].split(%r{[-/]}).first
+    end
+
+    def self.metadata
+      @metadata ||= Blacksmith::Modulefile.new.metadata
+    end
+  end
+
+  task "release:porcelain:changelog" do
+    # This is taken from lib/github_changelog_generator/task
+    # The generator cannot be used because we want to lazyly evaluate
+    # GCGConfig.user which might be overrider in the module Rakefile.
+    options = GitHubChangelogGenerator::Parser.default_options
+    options[:user] = GCGConfig.user
+    options[:project] = GCGConfig.metadata['name']
+    options[:future_release] = "v#{GCGConfig.metadata['version']}" if GCGConfig.metadata['version'] =~ /^\d+\.\d+.\d+$/
```

Comment:
Pretty sure not all modules use a `v` prefix. Ideally it's configurable in `GCGConfig`.

Also, if you don't use the match object then in modern Ruby it's preferred to use `.match?`.
```suggestion
    options[:future_release] = "v#{GCGConfig.metadata['version']}" if GCGConfig.metadata['version'].match?(/^\d+\.\d+.\d+$/)
```

Reply:
> Pretty sure not all modules use a `v` prefix. Ideally it's configurable in `GCGConfig`.

I used to exclude the `v` because I like tag to be the semver version, but this is unfortunately somewhat broken with GCG and I gave up. As far as I can see, we use a `v` in the modules tags, and the issues is only when GCG create the links to pages references containing tag that do not exist: if you mix `v1.2.3` and `1.2.3` in your existing tags, GCG generates the good url.  So not sure it's worth it to have a way to tune something that's broken.

Reply:
out of 170 modules managed by modulesync on my machine at $WORK, only these do not have a 'v' at the beginning of their last tag:

* puppet-augeasproviders_mounttab
* puppet-augeasproviders_nagios
* puppet-augeasproviders_puppet
* puppet-augeasproviders_sysctl
* puppet-module
* puppet-puppet_summary

These module have not seen a new version recently, maybe we can "standardize" on version numbers with a leading 'v'?

Reply:
In Foreman we don't have it, and I'd love to standardize the ecosystem in some way, so I'd prefer to the conditional part.

Reply:
Thinking more about it: tools often have a "tag pattern". That could default to `v%s` and some modules could override it.

Reply:
That makes sense :+1:

Reply:
Did this in a new commit (with the other part in https://github.com/voxpupuli/modulesync_config/pull/789). While doing this I realized that the pattern did not match the version we produce after creating a release ("<released-version.minor+1>-rc0", so I though we could remove it.

Reply:
The idea behind NOT matching rc0 is that we end up generating no future version. That means it ends up being "Unreleased" in the generated changelog. Though I'm not sure if this actually matches the real world.

Reply:
Got it! :+1: I amended my last commit accordingly